bin/yb-ctl

```diff
@@ -88,10 +90,25 @@ DAEMON_TYPES = [
 MAX_WAIT_ITERS = 20
 SLEEP_TIME_IN_SEC = 1
 MAX_WAIT_SECONDS = 45
+DEPRECATED_DEFAULT_NUM_DRIVES = 2
+DEFAULT_REPLICATION_FACTOR = 1
 
 YB_POSTGRES_PORT = 5433
 
 
+def retry_fn_with_timeout(fn, timeout_sec=None):
+    if timeout_sec is None:
+        timeout_sec = MAX_WAIT_SECONDS
+    start_time = time.time()
+    wait_count = 0
+    while wait_count < MAX_WAIT_ITERS and (time.time() - start_time <= timeout_sec):
```

Comment:
I think we should remove MAX_WAIT_ITERS and just use timeout_sec. Using MAX_WAIT_ITERS will end up not honoring timeout_sec. For example, if user's setup is slow and she increases timeout to X, it's likely that we'll timeout even before we reach X seconds due to MAX_WAIT_ITERS.

Reply:
+1